### PR TITLE
Booking details developed having with test cases

### PR DESCRIPTION
**Pull request by dipto-itransition**
This PR introduces the Booking detail page, allowing users to navigate to a specific booking's details by clicking on the booking within the Bookings table's Row or Booking card.

### Changes Made

**Mocked API Endpoint**: Added a new endpoint for fetching booking details at /api/bookings/{booking_id}.
**BookingDetail Component**: Created a new component to display the booking details.
**Routing**: Implemented routing to allow navigation to the Booking detail page at /bookings/{booking_id}.

### BookingDetail Layout

**Title:** Displays the booking number and its status.

**Trip Details Card:**
- Carrier name
- Departure and arrival information
- Booking date

**Purchaser Card:**
- Name
- Email
- Phone
- Address

**Payment Card:**
- Total price
- Retailer name and code
- Payment method

**Extras:**
- Placeholder cards with 'No extras' text for future use.

**Passengers List:**
- Displays a list of cards with passenger details (name, age category, birth date, ticket number).
- Age categories displayed: Infant (0-3 years), Child (4-15 years), Adult (16-99 years).

### Testing

- Added unit tests to ensure proper rendering of the Booking detail page under different states (loading, error, and successful data fetch).
- Confirmed branch coverage and functionality meet expected criteria.

### Screenshots

image
image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-import Layout from './components/layout/layout';
-
 const App: React.FC = () => {
   return (
     <BrowserRouter>
       <Layout>
         <Routes>
           <Route path={ROUTES.HOME} element={<Home />} />
           <Route path={ROUTES.BOOKINGS} element={<Bookings />} />
+          <Route path="/bookings/:id" element={<BookingDetails />} />
```
Comment (alexeylos) [inline src/App.tsx]: Please add `/bookings/:id` to `ROUTES` constant and use it

File: src/components/booking-details/PurchaserCard.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { TripDetailsCardProps } from '@/types/booking';
+import { EnvironmentOutlined, MailOutlined, PhoneOutlined } from '@ant-design/icons';
+import { Card } from 'antd';
+import './PurchaserCard.less';
+
+const PurchaserCard = ({ booking }: TripDetailsCardProps) => {
+  console.log(booking);
```
Comment (alexeylos) [inline src/components/booking-details/PurchaserCard.tsx]: Please remove it

File: src/components/booking-details/TripDetailsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { ArrowRightOutlined } from '@ant-design/icons';
+import { Card } from 'antd';
+import dayjs from 'dayjs';
+import './TripDetailsCard.less';
+
+const TripDetailsCard = ({ booking }: TripDetailsCardProps) => {
+  return (
+    <div>
+      <Card title="Trip Details">
+        <p className="carrier-name">carrier name : {booking.carrier_name}</p>
```
Comment (alexeylos) [inline src/components/booking-details/TripDetailsCard.tsx]: Let's use capital for the first letter

File: src/components/booking-details/TripDetailsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <div className="divider"></div>
```
Comment (alexeylos) [inline src/components/booking-details/TripDetailsCard.tsx:12]: nit: You may also use Divider component from Ant

File: src/components/bookings/BookingListMobileView.test.tsx [hunk 5]
```diff
@@ -45,7 +51,12 @@ describe('BookingListMobileView', () => {
       isError: true,
     });
 
-    render(<BookingListMobileView />); // Assuming you handle errors inside the component
+    render(
+      <MemoryRouter>
+        {' '}
```
Comment (alexeylos) [inline src/components/bookings/BookingListMobileView.test.tsx]: Unnecessary space
And few more in this file

File: src/components/shared/HamburgerMenu.tsx [hunk 6]
```diff
@@ -26,7 +26,11 @@ const HamburgerMenu: React.FC = () => {
   };
 
   useEffect(() => {
-    setCurrentLocation(location.pathname);
+    if (location.pathname == '/bookings/B1') {
```
Comment (alexeylos) [inline src/components/shared/HamburgerMenu.tsx]: Why so?

File: src/pages/BookingDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    return <div>Error fetching data</div>;
+  }
+
+  return (
+    <div>
+      <Row gutter={[20, 20]}>
+        <Col xs={24} md={12} xl={8}>
+          <div className="trip-details-extra-info">
+            <TripDetailsCard booking={booking}></TripDetailsCard>
+            <ExtraInfo></ExtraInfo>
```
Comment (alexeylos) [inline src/pages/BookingDetails.tsx]: You may use shorthand `<ExtraInfo />`

File: src/pages/BookingDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+          </div>
+        </Col>
+        <Col xs={24} md={12} xl={16}>
+          <div className="trip-details-extra-info">
+            <div>
+              <Row gutter={[20, 20]}>
+                <Col xs={24} sm={12} md={24} lg={24} xl={12}>
+                  <PurchaserCard booking={booking}></PurchaserCard>
```
Comment (alexeylos) [inline src/pages/BookingDetails.tsx]: Same `<PurchaserCard booking={booking} />`
No need in closing tag if there are no children of the node

File: src/components/booking-details/PassengersListCard.test.tsx [hunk 9]
```diff
@@ -0,0 +1,73 @@
+import { BookingDetailsData } from '@/types/booking';
+import { render, screen } from '@testing-library/react';
+import dayjs from 'dayjs';
+import PassengersListCard from './PassengersListCard';
+
+const mockBooking: BookingDetailsData = {
```
Comment (alexeylos) [inline src/components/booking-details/PassengersListCard.test.tsx]: Please move it to `src/msw/bookingDetailsMockData.ts` and reuse in all other tests

File: src/components/booking-details/TripDetailsCard.tsx (hunk 4 above)
Comment (dipto-itransition) [reply src/components/booking-details/TripDetailsCard.tsx:12]: Thanks for your suggestion. but I just want to customize the divider, that's why I use this one!

File: src/components/bookings/BookingListMobileView.test.tsx (hunk 5 above)
Comment (dipto-itransition) [reply src/components/bookings/BookingListMobileView.test.tsx]: That's happening due to auto saving and formatting. I will be more careful from the next time

File: src/components/shared/HamburgerMenu.tsx (hunk 6 above)
Comment (dipto-itransition) [reply src/components/shared/HamburgerMenu.tsx]: To keep the user's navigation active on booking page. although I forget to make it dynamic. 
updated one :


```
useEffect(() => {
    if (/^\/bookings\/[^/]+$/.test(location.pathname)) {
      setCurrentLocation('/bookings');
    } else {
      setCurrentLocation(location.pathname);
    }
  }, [location]);
```

File: src/constants/common.ts [hunk 10]
```diff
@@ -1,4 +1,5 @@
 export const ROUTES = {
   HOME: '/',
   BOOKINGS: '/bookings',
+  BOOKING_DETAILS: (id: string) => `/bookings/${id}`,
```
Comment (alexeylos) [inline src/constants/common.ts]: No need to do that. React router resolves dynamic routes under the hood. Just use  `BOOKING_DETAILS: '/bookings/:id'`

File: src/App.tsx [hunk 11]
```diff
@@ -1,17 +1,18 @@
...
-import Layout from './components/layout/layout';
-
 const App: React.FC = () => {
   return (
     <BrowserRouter>
       <Layout>
         <Routes>
           <Route path={ROUTES.HOME} element={<Home />} />
           <Route path={ROUTES.BOOKINGS} element={<Bookings />} />
+          <Route path={ROUTES.BOOKING_DETAILS(':id')} element={<BookingDetails />} />
```
Comment (alexeylos) [inline src/App.tsx]: And then just use `path={ROUTES.BOOKING_DETAILS}` here

File: src/components/reusable/booking/BookingItemCard.tsx [hunk 12]
```diff
@@ -17,16 +19,21 @@ const BookingItemCard: React.FC<Booking> = ({
...
 }) => {
+  const navigate = useNavigate();
+  const onClickNavigate = (id: string) => {
+    navigate(`/bookings/${id}`);
+  };
+
   return (
     <>
-      <Card>
+      <Card onClick={() => onClickNavigate(id)}>
```
Comment (alexeylos) [inline src/components/reusable/booking/BookingItemCard.tsx]: Actually there's no need to pass `id` here. You can get it from props
```
const onClickNavigate = useCallback(() => {
    navigate(`/bookings/${id}`);
  }, [id]);
```
And then just
```
<Card onClick={onClickNavigate}>
```

**Review by alexeylos — APPROVED** [review]

**Comment by dipto-itransition** [comment]
Introduces the Booking detail page, allowing users to navigate to a specific booking's details

**Pull request merged**